beamer/util.py

```diff
@@ -176,10 +176,14 @@ def is_valid_pair(
         source_token_address: ChecksumAddress,
         target_chain_id: ChainId,
         target_token_address: ChecksumAddress,
+        current_target_chain_id: ChainId,
```

Comment:
During the whole runtime, `current_target_chain_id` will be a constant value. I'd rather add this to the TokenMatchChecker once (during init) instead of passing to the function everytime we call it. Seems a bit inefficient.